src/ingest-pipeline/misc/tools/survey.py

```diff
@@ -220,7 +221,7 @@ def __init__(self, prop_dct, entity_factory):
 
     @property
     def donor_uuid(self):
-        if self._donor_uuid is None:
+        if self._donor_uuid is None and self.parent_uuids:
```

Comment:
This change results in donor_uuid being None if self.parent_uuids is false, and it really surprises me that that doesn't break anything.  What comes out on the donor_uuid line in the final spreadsheet when this happens?

Reply:
For all the data we already have, there is always a parent_uuid which makes the donor_uuid not None. Only the bogus ones are failing into this category where _donor_uuid is none.

Reply:
Yes, and I agree that the problem will only occur until Bill's team fixes the bug in their endpoint, but what does the failure mode look like?  I thought it was going to be necessary to create a special entity that would play a role like <NO-SUCH-ENTITY> in order to keep things running smoothly.

Reply:
It would just skip most of the information for the bogus entities and keep on processing the good ones (but saves enough to be showed in the sheet if you want to identify was is going on with them), you can't actually create a 'special' entity, because that would need to exist in the DB so the entity-search can retrieve something for it. Not sure if it is worth the extra effort. Did I missed something there?

Reply:
My thought was to insert the special entity in the cache of entity-search results so it wouldn't have to go all the way to entity-api, but I agree that it's not necessary.